mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/application/chess/rules/MovesTest.kt

```diff
@@ -1,22 +1,25 @@
 package ch.epfl.sdp.mobile.test.application.chess.rules
 
-import ch.epfl.sdp.mobile.application.chess.Delta
-import ch.epfl.sdp.mobile.application.chess.Piece
-import ch.epfl.sdp.mobile.application.chess.Position
-import ch.epfl.sdp.mobile.application.chess.Rank.Pawn
+import ch.epfl.sdp.mobile.application.chess.*
+import ch.epfl.sdp.mobile.application.chess.Rank.*
 import ch.epfl.sdp.mobile.application.chess.implementation.PersistentPieceIdentifier
 import ch.epfl.sdp.mobile.application.chess.implementation.buildBoard
 import ch.epfl.sdp.mobile.application.chess.implementation.emptyBoard
 import ch.epfl.sdp.mobile.application.chess.rules.*
 import ch.epfl.sdp.mobile.application.chess.rules.Role.Adversary
 import ch.epfl.sdp.mobile.application.chess.rules.Role.Allied
+import ch.epfl.sdp.mobile.test.application.chess.buildBoardWithHistory
 import com.google.common.truth.Truth.assertThat
 import org.junit.Test
 
 class MovesTest {
 
+  private val adversaryKing = Piece(Adversary, King, PersistentPieceIdentifier(0))
   private val adversaryPawn = Piece(Adversary, Pawn, PersistentPieceIdentifier(0))
+  private val adversaryRook = Piece(Adversary, Rook, PersistentPieceIdentifier(0))
```

Comment:
[Question] Why is it the case that different Pieces have the same `PersistentPieceIdentifier(0)` namley 0.

Reply:
We've had a discussion about this with @matt989253 : https://github.com/epfl-SDP/android/pull/147#discussion_r833223998.